src/main/java/org/wildfly/security/auth/client/ElytronXmlParser.java

```diff
@@ -965,8 +983,16 @@ private static int setBit(int var, int bit) {
                         break;
                     }
                     case "oauth2-bearer-token": {
-                        CredentialSource credentialStore = parseOAuth2BearerTokenType(reader, expectedNamespace);
-                        function = andThenOp(function, credentialSource -> credentialSource.with(credentialStore));
+                        CredentialSource oauthCredentialSource = parseOAuth2BearerTokenType(reader, expectedNamespace);
+                        function = andThenOp(function, credentialSource -> credentialSource.with(oauthCredentialSource));
+                        break;
+                    }
+                    case "local-kerberos": {
+                        if ( ! KNOWN_NAMESPACES.get(expectedNamespace).isAtLeast(1, 1)) {
```

Comment:
If we convert the whole namespace and version into a single enum we can just pass that around as the expected namespace so we don't need to retrieve it from the map for each check.